standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreTServerSocket.java

```diff
@@ -29,8 +32,32 @@
  * accepted sockets.
  *
  */
-public class TServerSocketKeepAlive extends TServerSocket {
-  public TServerSocketKeepAlive(TServerSocket serverSocket) throws TTransportException {
+public class MetastoreTServerSocket extends TServerSocket {
+  private final Configuration configuration;
+
+  public MetastoreTServerSocket(TServerSocket serverSocket,
```

Comment:
why pass conf, instead of tcpKeepAlive var and maxThriftMessageSize?

Reply:
1. for changing the limit in test easily, as this class is initialized on HMS startup; 2. through configuration it can make customize the socket easily in the future, such as configuring the socket read timeout